pkg/config/polling_manager.go

```diff
@@ -182,48 +197,58 @@ func (cm *PollingProjectConfigManager) Start(ctx context.Context) {
 	pollingProjectConfigManager := PollingProjectConfigManager{
-		notificationCenter:  registry.GetNotificationCenter(sdkKey),
-		pollingInterval:     DefaultPollingInterval,
-		requester:           utils.NewHTTPRequester(logging.GetLogger(sdkKey, "HTTPRequester")),
-		datafileURLTemplate: DatafileURLTemplate,
-		sdkKey:              sdkKey,
-		logger:              logging.GetLogger(sdkKey, "PollingProjectConfigManager"),
+		notificationCenter: registry.GetNotificationCenter(sdkKey),
+		pollingInterval:    DefaultPollingInterval,
+		requester:          utils.NewHTTPRequester(logging.GetLogger(sdkKey, "HTTPRequester")),
+		sdkKey:             sdkKey,
+		logger:             logger,
 	}
 
-	for _, opt := range pollingMangerOptions {
+	for _, opt := range configOptions {
 		opt(&pollingProjectConfigManager)
 	}
 
+	if pollingProjectConfigManager.datafileURLTemplate == "" {
+		if pollingProjectConfigManager.datafileAccessToken != "" {
+			pollingProjectConfigManager.datafileURLTemplate = AuthDatafileURLTemplate
+		} else {
+			pollingProjectConfigManager.datafileURLTemplate = DatafileURLTemplate
+		}
+	}
+	pollingProjectConfigManager.setDatafileAccessTokenRequest()
+	return &pollingProjectConfigManager
+}
+
+// NewPollingProjectConfigManager returns an instance of the polling config manager with the customized configuration
+func NewPollingProjectConfigManager(sdkKey string, pollingMangerOptions ...OptionFunc) *PollingProjectConfigManager {
+
+	pollingProjectConfigManager := newConfigManager(sdkKey, logging.GetLogger(sdkKey, "PollingProjectConfigManager"), pollingMangerOptions...)
+
 	if len(pollingProjectConfigManager.initDatafile) > 0 {
 		pollingProjectConfigManager.setInitialDatafile(pollingProjectConfigManager.initDatafile)
 	} else {
 		pollingProjectConfigManager.SyncConfig() // initial poll
 	}
-	return &pollingProjectConfigManager
+	return pollingProjectConfigManager
 }
 
 // NewAsyncPollingProjectConfigManager returns an instance of the async polling config manager with the customized configuration
 func NewAsyncPollingProjectConfigManager(sdkKey string, pollingMangerOptions ...OptionFunc) *PollingProjectConfigManager {
 
-	pollingProjectConfigManager := PollingProjectConfigManager{
-		notificationCenter:  registry.GetNotificationCenter(sdkKey),
-		pollingInterval:     DefaultPollingInterval,
-		requester:           utils.NewHTTPRequester(logging.GetLogger(sdkKey, "HTTPRequester")),
-		datafileURLTemplate: DatafileURLTemplate,
-		sdkKey:              sdkKey,
-		logger:              logging.GetLogger(sdkKey, "PollingProjectConfigManager"),
-	}
-
-	for _, opt := range pollingMangerOptions {
-		opt(&pollingProjectConfigManager)
-	}
-
+	pollingProjectConfigManager := newConfigManager(sdkKey, logging.GetLogger(sdkKey, "PollingProjectConfigManager"), pollingMangerOptions...)
 	pollingProjectConfigManager.setInitialDatafile(pollingProjectConfigManager.initDatafile)
```

Comment:
Unrelated to your PR: it is suspicious that, in `NewPollingProjectConfigManager`, this `setInitialDatafile` call is surrounded by `if len(pollingProjectConfigManager.initDatafile) > 0 {`, while here, it is not surrounded by an `if`. Do we need this check or not?

Reply:
we can, but we don't have to. I will make this change, but the end result will be the same. in NewPollingProjectConfigManager it is really if/else , here we don't have if/else and we try loading initial file first. if we have nil for initial file, then it will not load but we will see the error in the log. So the end result will be the same, but it will be cleaner to check it first. good catch